vtex/faststore
# chore(core): optimize graphql from schema file

## What's the purpose of this pull request?

This PR adds the ability to generate GraphQL types & optimizations without a running development server. It also adds the ability for starters to generate these GraphQL types & optimizations, making it possible for them to actually use their API customizations.

## How it works?

A new `generate-graphql` command was added to the FastStore CLI, which makes it possible to generate the GraphQL types & optimizations from the starter. This generation is made by creating a schema file beforehand and using this schema as the basis for the generation. Previously it used the running graphql dev server, which was an issue considering the development flow in the starters.

## How to test it?

https://github.com/vtex-sites/starter.store/pull/180

Use this starter. Open a development server or generate a build with it. See optimizations and types being generated inside the `.faststore/@generated` folder. Also, use the new generate-graphql command by running `yarn generate`.

### Starters Deploy Preview

https://github.com/vtex-sites/starter.store/pull/180
